From 55c984960f80e6a62d8e11195fb772f37b9b6985 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Tue, 7 Nov 2017 18:33:52 +0000
Subject: [PATCH] Update Makefile for building with a default GOPATH

Using the template from cri-o, update the Makefile to provide a
default GOPATH and create the structure to allow that GOPATH to build
successfully.
---
 Makefile | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/Makefile b/Makefile
index 4869882..a21625b 100644
--- a/Makefile
+++ b/Makefile
@@ -15,8 +15,15 @@
 GO ?= go
 PROJECT := github.com/kubernetes-incubator/cri-tools
 BINDIR := /usr/local/bin
+ifeq ($(GOPATH),)
+export GOPATH := $(CURDIR)/_output
+unexport GOBIN
+endif
 GOBINDIR := $(word 1,$(subst :, ,$(GOPATH)))
 PATH := $(GOBINDIR)/bin:$(PATH)
+GOPKGDIR := $(GOPATH)/src/$(PROJECT)
+GOPKGBASEDIR := $(shell dirname "$(GOPKGDIR)")
+
 
 all: binaries
 
@@ -28,6 +35,10 @@ help:
 	@echo " * 'clean' - Clean artifacts."
 
 check-gopath:
+ifeq ("$(wildcard $(GOPKGDIR))","")
+	mkdir -p "$(GOPKGBASEDIR)"
+	ln -s "$(CURDIR)" "$(GOPKGBASEDIR)/cri-tools"
+endif
 ifndef GOPATH
 	$(error GOPATH is not set)
 endif
-- 
2.15.0

